### ISSUE TITLE
Export and Alias Revisited

### ISSUE THREAD

**Issue by ericspod** [Project-MONAI/MONAI#240]
**Is your feature request related to a problem? Please describe.**
Export and alias should be re-introduced with semantics to support defining actual aliases in the module where decorated definitions are found. The `@export` decorator should then work to correctly export these aliases.

**Describe the solution you'd like**
Slight change to export and alias decorators.

**Comment by wyli** [comment]
I agree that the current way of exposing public APIs (importing in `__init__.py` such as https://github.com/Project-MONAI/MONAI/blob/master/monai/transforms/__init__.py)  could be improved, but it's unclear to me whether the decorator approach is the way forward. 

I'm wondering if it is possible to list a few alternative approaches and have a discussion here?

it seems some of the well-known python codebases go for a simple `__init__.py` approach, for example:  flask https://github.com/pallets/flask/blob/master/src/flask/__init__.py

**Comment by ericspod** [comment]
It's the standard way to put things into `__init__.py` manually, if you use `import *` you need to then use `__ALL__` to control what members are exported. It all comes down to boilerplate which would be nice to avoid. A proposed solution is PR #241.

**Comment by wyli** [comment]
thanks for the PR, I feel it's an important design decision to make. We should go for the standard approach or start voting.

apart from that, if part of the intention is to resolve the `import *` issue, why not setting the modules' `__all__` in the export method?

**Comment by Nic-Ma** [comment]
Hi @ericspod ,

As your PR was merged, can we close this PR now?
Thanks.
